crates/executor/executor.rs

```diff
@@ -29,6 +28,32 @@ pub enum ExecuteResult {
 
 pub type OnCached = dyn Fn() -> ExecuteResult;
 
+pub fn generate_interaction_context(tx: &GQLNodeInterface) -> Value {
+  let interaction_context = serde_json::json!({
+            "transaction": {
+              "id": tx.id,
+              "owner": tx.owner.address,
+              "target": tx.recipient.to_owned().unwrap_or_else(|| String::from("null")),
+              "quantity": tx.quantity.to_owned().unwrap_or_else(|| GQLAmountInterface {
+    winston: Some(String::new()),
+    ar: Some(String::new())
+  }).winston.unwrap(),
+              "reward": tx.fee.to_owned().unwrap_or_else(|| GQLAmountInterface {
+    winston: Some(String::new()),
+    ar: Some(String::new())
+  }).winston.unwrap(),
+              "tags": tx.tags
+            },
+            "block": {
+              "indep_hash":  tx.block.id,
+              "height": tx.block.height,
+              "timestamp": tx.block.timestamp
+            }
+          });
+
```

Comment:
This can be typed:
```rust
#[derive(Deserialize)]
pub struct ContractTx {
  pub id: String,
  pub owner: String,
  pub tags: Vec<Tag>,
  pub target: String,
  pub quantity: String,
  pub reward: String,
}

#[derive(Deserialize)]
pub struct ContractBlock {
  pub height: usize,
  pub indep_hash: String,
  pub timestamp: usize,
}

#[derive(Deserialize)]
pub struct ContractInfo {
  pub transaction: ContractTx,
  pub block: ContractBlock,
}
```